src/courtroom.cpp

```diff
@@ -1573,45 +1573,22 @@ void Courtroom::play_sfx()
 
 void Courtroom::set_text_color()
 {
+  QMap<Color, QStringList> all_colors = ao_app->get_chatmessage_colors();
+  Color color = (Color)m_chatmessage[CMTextColor].toInt();
 
-  switch (m_chatmessage[CMTextColor].toInt())
+  QString color_code;
+  if (all_colors.contains(color))
   {
-  case CGreen:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(101, 200, 86);
-    break;
-  case CRed:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(186, 21, 24);
-    break;
-  case COrange:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(213, 89, 0);
-    break;
-  case CBlue:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(21, 136, 200);
-    break;
-  case CYellow:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(231, 206, 78);
-    break;
-  case CPurple:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(247, 118, 253);
-    break;
-  case CPink:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(218, 124, 128);
-    break;
-  default:
-    qDebug() << "W: undefined text color: " << m_chatmessage[CMTextColor];
-    [[fallthrough]];
-  case CWhite:
-    ui_vp_message->setStyleSheet("background-color: rgba(0, 0, 0, 0)");
-    m_base_string_color.setRgb(213, 213, 213);
-    break;
+    color_code = all_colors[color][1];
+  }
+  else
+  {
+    // Fallback color in case some other client sent a message with an out of bounds color
+    color_code = all_colors[CWhite][1];
```

Comment:
You shouldn't use it like an array, use a struct instead.

```cpp
struct ColorInfo
{
  QString name;
  QString code;
}

Reply:
A struct is easier to use and is a lot more straightforward: want the color code? You can just pick it up from the struct. Instead, what we're doing here is have this weird QStringList and we're using it like an array? What if it changes and we later add another field inbetween name and code?